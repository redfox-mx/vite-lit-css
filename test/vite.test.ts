import { resolve } from "node:path";
import { describe, expect, it } from "vitest";
import { build } from "vite";
import glob from "fast-glob";
import litCss from "../src";

async function getCode(file: string) {
  const bundle = (await build({
    plugins: [
      litCss()
    ],
    build: {
      lib: {
        entry: file,
        formats: ["es"],
      },
      rollupOptions: {
        external: [/^lit/, /^@lit/],
      },
    },
  })) as any;

  const [output] = bundle
  return output.output
    .map((file: any) => {
      if (file.type === "chunk") {
        return `//${file.fileName}\n${file.code}`;
      } else {
        return file.fileName;
      }
    })
    .join("\n");
}

describe("litcss plugin", () => {
  describe("fixtures", async () => {
    const root = resolve(__dirname, "..");
    const files = await glob("test/fixtures/*.ts", {
      cwd: root,
      onlyFiles: true,
    });

    for (const file of files) {
      describe(file.replaceAll("\\", "/"), async () => {
        const filepath = resolve(root, file);

        for (const isProduction of [true, false]) {
          it(`isProduction is ${isProduction}`, async () => {
            process.env.NODE_ENV = isProduction ? "production" : "development";

            const code = await getCode(file);
            expect(
              code.replaceAll(JSON.stringify(filepath), "'#FILE#'")
            ).toMatchSnapshot();
          });
        }
      });
    }
  });
});
