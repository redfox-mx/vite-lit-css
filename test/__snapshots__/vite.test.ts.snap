// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`litcss plugin > fixtures > test/fixtures/default-import.ts > isProduction is false 1`] = `
"//vite-plugin-lit-css.js
import { css as o } from \\"lit\\";
const e = o\`:host{color:red}#xss{content:\\\\\\"\\\\\${console.log(window)}\\\\\\"}.hover\\\\\\\\:underline{text-decoration:underline}.text-\\\\\\\\[\\\\\\\\#50d71e\\\\\\\\]{color:#50d71e}\\\\n\`, t = e;
export {
  t as styles
};
"
`;

exports[`litcss plugin > fixtures > test/fixtures/default-import.ts > isProduction is true 1`] = `
"//vite-plugin-lit-css.js
import { css as o } from \\"lit\\";
const e = o\`:host{color:red}#xss{content:\\\\\\"\\\\\${console.log(window)}\\\\\\"}.hover\\\\\\\\:underline{text-decoration:underline}.text-\\\\\\\\[\\\\\\\\#50d71e\\\\\\\\]{color:#50d71e}\\\\n\`, t = e;
export {
  t as styles
};
"
`;

exports[`litcss plugin > fixtures > test/fixtures/query-import.ts > isProduction is false 1`] = `
"//vite-plugin-lit-css.js
import { css as o } from \\"lit\\";
const e = o\`:host{color:red}#xss{content:\\\\\\"\\\\\${console.log(window)}\\\\\\"}.hover\\\\\\\\:underline{text-decoration:underline}.text-\\\\\\\\[\\\\\\\\#50d71e\\\\\\\\]{color:#50d71e}\\\\n\`, t = e;
export {
  t as styles
};
"
`;

exports[`litcss plugin > fixtures > test/fixtures/query-import.ts > isProduction is true 1`] = `
"//vite-plugin-lit-css.js
import { css as o } from \\"lit\\";
const e = o\`:host{color:red}#xss{content:\\\\\\"\\\\\${console.log(window)}\\\\\\"}.hover\\\\\\\\:underline{text-decoration:underline}.text-\\\\\\\\[\\\\\\\\#50d71e\\\\\\\\]{color:#50d71e}\\\\n\`, t = e;
export {
  t as styles
};
"
`;
